src/app/services/distance.service.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DistanceService {
+  //public dict? : Map<string, { latitude: number; longitude: number }>;
+  public dict = {};
+  constructor(private http: HttpClient) {
+    this.makeDict();
+  }
+  public makeDict() {
+    this.getInfo().subscribe((data) => {
+      const list = data.split('\n');
+      list.shift();
+      list.forEach((e) => {
+        const pair = e.split(`,"`);
+
+        this.dict[pair[0]] = `"${pair[1]}`;
+      });
+    });
+  }
+  getInfo() {
+    return this.http.get('././assets/uszips2.csv', { responseType: 'text' });
+  }
+  getCoord(zip: string) {
```

Comment:
This appears to be the "real" public API and should have a type specified for its return. It appears to be a string that's really a JSON object?